llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -1454,8 +1455,20 @@ static bool checkAndReplaceCondition(
       return ShouldReplace;
     });
     NumCondsRemoved++;
-    if (Cmp->use_empty())
+    if (Cmp->use_empty()) {
```

Comment:
Hmmm.. .I think we probably want to do the debug update outside of the `if (Cmp->use_empty())` block (unconditionally)? Otherwise we're only updating debug info if there are no other non-dbg uses. But IMO we should update the debug info regardless, because we know the value on this code path. 

(I'm not sure there's a clear correct answer, happy to discuss further).